sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamTest.scala

```diff
@@ -669,55 +658,48 @@ trait StreamTest extends QueryTest with SharedSQLContext with Timeouts {
     }
   }
 
-
-  class QueryStatusCollector extends StreamingQueryListener {
+  /** Collects events from the StreamingQueryListener for testing */
+  class EventCollector extends StreamingQueryListener {
```

Comment:
I don't think this needs to be an inner class of `StreamTest`.  This file is pretty long/complicated as is.